src/Tes.Runner.Test/ExecutorTests.cs

```diff
@@ -79,6 +92,33 @@ public async Task DownloadInputsAsync_InputProvided_StartSuccessEventsAreCreated
             eventsPublisherMock.Verify(p => p.PublishDownloadEndEventAsync(It.IsAny<NodeTask>(), 0, 0, EventsPublisher.SuccessStatus, string.Empty), Times.Once);
         }
 
+        [TestMethod]
+        public async Task DownloadInputsAsync_InputsProvided_NumberOfFilesInEndEventMatchesTheInputCount()
+        {
+            var inputs = new List<DownloadInfo>
+            {
+                new DownloadInfo( FullFilePath: "/mnt/data/input1.txt", SourceUrl: new Uri("https://test.blob.core.windows.net/test/input1.txt"))
+            };
+
+            nodeTask.Inputs = new List<FileInput>
+            {
+                new FileInput()
+                {
+                    Path = "mnt/data/input",
+                    SourceUrl = "https://test.blob.core.windows.net/test/input1.txt"
+                }
+            };
+
+            blobDownloaderMock.Setup(r => r.DownloadAsync(It.IsAny<List<DownloadInfo>>()))
+                .ReturnsAsync(0);
+
+            fileOperationResolverMock.Setup(r => r.ResolveInputsAsync()).ReturnsAsync(inputs);
+            var result = await executor.DownloadInputsAsync(blobPipelineOptions);
+            Assert.AreEqual(Executor.ZeroBytesTransferred, result);
+            eventsPublisherMock.Verify(p => p.PublishDownloadStartEventAsync(It.IsAny<NodeTask>()), Times.Once);
+            eventsPublisherMock.Verify(p => p.PublishDownloadEndEventAsync(It.IsAny<NodeTask>(), nodeTask.Inputs.Count, 0, EventsPublisher.SuccessStatus, string.Empty), Times.Once);
```

Comment:
Maybe add another test or modify this one, to test a non-zero number of inputs and assert the download count, since that was the original bug?

Reply:
Hmm this is what this test is asserting. It verifies that the event is created with the correct number of files (nodeTask.Inputs.Count).